lib/TPP/ConvertLinalgToTpp.cpp

```diff
@@ -35,33 +35,51 @@ struct ConvertGenericOpToTpp : public OpRewritePattern<linalg::GenericOp> {
   LogicalResult rewriteToTppOp(linalg::GenericOp linalgOp,
                                PatternRewriter &rewriter) const {
     SmallVector<Value> operands;
+    if (tpp::utils::isTppZero(linalgOp, &operands)) {
+      assert((operands.size() > 0 && operands.size() <= 2) &&
+             "tpp.zero expects one or two operands");
+
+      // Only take the output, the other operand might be a constant.
+      rewriter.replaceOpWithNewOp<tpp::ZeroOp>(linalgOp,
+                                               operands[operands.size() - 1],
+                                               operands[operands.size() - 1]);
+      return success();
+    }
+
     if (tpp::utils::isTppIdentity(linalgOp, &operands)) {
-      assert(operands.size() == 2 && "Expect two operands");
+      if (operands.size() != 2)
```

Comment:
Same comment here, the `operands.size()` should be inside the match above and just "not match" and continue.

Reply:
Similar to all other existing ones... We shouldn't fail the compilation process if we match only half-way through, we should just not match and continue.

Reply:
I agree the internal if check shouldn't be there. Its place is in the util matcher. I'd still keep the asserts just as a double check. If the assert ever triggers, it means the matcher is too generic.

Reply:
We can do this at another commit, but we *have* to move to that style or we'll keep generating broken code.